### PR TITLE
4-modules

### PR DESCRIPTION
**Pull request by BuzRom**
https://buzrom.github.io/frontend-course-2021/homeworks/roman.buzukov_BuzRom/4-modules/

### CODE REVIEW

File: homeworks/roman.buzukov_BuzRom/4-modules/js/time.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export const isMonthLong = (date) => {
+    const splitDate = date.split('-');
+    const month = splitDate[1];
+    const year = splitDate[0];
+    return new Date(year, month, 0).getDate() === 31;
+};
+
+export const shortestWeekDaysNumber = (date) => {
+    const weeks = getWeeksInMonth(date);
+    return (weeks[0].length < weeks[weeks.length - 1].length)
```
Comment (electrovladyslav) [inline homeworks/roman.buzukov_BuzRom/4-modules/js/time.js:42]: same here
Store `store weeks[0].length` and `weeks[weeks.length - 1].length` to the proper named vars

Comment (electrovladyslav) [reply homeworks/roman.buzukov_BuzRom/4-modules/js/time.js:42]: Or use `Math.max`))))

**Comment by electrovladyslav** [comment]
> https://buzrom.github.io/frontend-course-2021/homeworks/roman.buzukov_BuzRom/4-modules/
> 
> I didn't get how to do asterisk task with given two functions. In my opinion, the easiest way is to use 'timezone'.

At least you should realize those two functions `subtractHours` and `addHours`.
And further, you should use it to calculate the time for a given city based on local time.
Sorry, but without it, I can't count the task with an asterisk as done.

File: homeworks/roman.buzukov_BuzRom/4-modules/index.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    <body>
+        <div class="container" data-container>
+            <ul class="container__list">
+                <li class="container__list-item item">
+                    <p class="item__text">Date:</p>
+                    <input class="item__input" type="date" data-userMonth>
+                </li>
+                <li class="container__list-item item">
+                    <select class="container__list-select" data-daySelector>
```
Comment (electrovladyslav) [inline homeworks/roman.buzukov_BuzRom/4-modules/index.html:20]: Super

File: homeworks/roman.buzukov_BuzRom/4-modules/index.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        <title>Document</title>
+        <link rel="stylesheet" href="css/style.css">
+    </head>
+
+    <body>
+        <div class="container" data-container>
+            <ul class="container__list">
+                <li class="container__list-item item">
+                    <p class="item__text">Date:</p>
+                    <input class="item__input" type="date" data-userMonth>
```
Comment (electrovladyslav) [inline homeworks/roman.buzukov_BuzRom/4-modules/index.html:17]: type=date - 👍

**Review by electrovladyslav — CHANGES_REQUESTED** [review]

File: homeworks/roman.buzukov_BuzRom/4-modules/css/style.css [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+body {
+  font-family: "Montserrat", sans-serif;
+  background-color: rgba(230, 239, 253, 1);
+}
+
+.container {
+  max-width: 1200px;
+  height: calc(100vh - 100px);
```
[the remark this answers is not in the document]
Comment (electrovladyslav) [reply homeworks/roman.buzukov_BuzRom/4-modules/css/style.css:16]: Same here 
image

File: homeworks/roman.buzukov_BuzRom/3-async/css/style.css [hunk 5]
```diff
@@ -1,193 +1,192 @@
-@import url("https://fonts.googleapis.com/css2?family=Montserrat:wght@400;500;900&display=swap");
```
[the remark this answers is not in the document]
Comment (BuzRom) [reply homeworks/roman.buzukov_BuzRom/3-async/css/style.css:1]: This file from previous task '3-async'. I found one small bug and fix it.

[the remark this answers is not in the document]
Comment (electrovladyslav) [reply homeworks/roman.buzukov_BuzRom/3-async/css/style.css:1]: Ok then

File: homeworks/roman.buzukov_BuzRom/4-modules/css/style.css (hunk 4 above)
[the remark this answers is not in the document]
Comment (electrovladyslav) [reply homeworks/roman.buzukov_BuzRom/4-modules/css/style.css]: Cool 
image

File: homeworks/roman.buzukov_BuzRom/4-modules/js/asterisk.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+export function addHours(date, hours) {
+    return new Date(new Date(date).setHours(date.getHours() + hours)).toLocaleTimeString().split(':').join(' : ');
```
Comment (electrovladyslav) [inline homeworks/roman.buzukov_BuzRom/4-modules/js/asterisk.js:2]: No need to make two Date objects. You can create one, setHours for it, and work with it further.

Comment (electrovladyslav) [reply homeworks/roman.buzukov_BuzRom/4-modules/js/asterisk.js:2]: Also, this line is too long and difficult to understand. Split it into some simpler lines.

File: homeworks/roman.buzukov_BuzRom/4-modules/js/time.js [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    const month = splitDate[1];
+    const year = splitDate[0];
+    return new Date(year, month, 0).getDate() === 31;
+};
+
+export const shortestWeekDaysNumber = (date) => {
+    const weeks = getWeeksInMonth(date);
+    const firstWeek = weeks[0].length;
+    const lastWeek = weeks[weeks.length - 1].length;
+    return (firstWeek < lastWeek)
```
Comment (electrovladyslav) [inline homeworks/roman.buzukov_BuzRom/4-modules/js/time.js:44]: Ok, it's much better now. Last [time](https://github.com/MastersAcademy/frontend-course-2021/pull/141#discussion_r748477612) I've mismatched: the `Math.min`  could be a choice
```
return Math.min(firstWeek, lastWeek)
```

**Review by electrovladyslav — CHANGES_REQUESTED** [review]

**Review by electrovladyslav — APPROVED** [review]

**Comment by electrovladyslav** [comment]
Super, thanx, for your job

**Pull request merged**